src/vm/tests/contracts.rs

```diff
@@ -486,6 +500,104 @@ fn test_contract_caller(owned_env: &mut OwnedEnvironment) {
     }
 }
 
+fn tx_sponsor_contract_asserts(env: &mut Environment, sponsor: Option<PrincipalData>) {
+    let sponsor = match sponsor {
+        None => Value::none(),
+        Some(p) => Value::some(Value::Principal(p)).unwrap(),
+    };
+    assert_eq!(
+        env.execute_contract(
+            &QualifiedContractIdentifier::local("contract-a").unwrap(),
+            "get-sponsor",
+            &vec![],
+            false
+        )
+        .unwrap(),
+        Value::list_from(vec![sponsor.clone()]).unwrap()
+    );
+    assert_eq!(
+        env.execute_contract(
+            &QualifiedContractIdentifier::local("contract-b").unwrap(),
+            "as-contract-get-sponsor",
+            &vec![],
+            false
+        )
+        .unwrap(),
+        Value::list_from(vec![sponsor.clone()]).unwrap()
+    );
+    assert_eq!(
+        env.execute_contract(
+            &QualifiedContractIdentifier::local("contract-b").unwrap(),
+            "cc-get-sponsor",
+            &vec![],
+            false
+        )
+        .unwrap(),
+        Value::list_from(vec![sponsor.clone()]).unwrap()
+    );
+    assert_eq!(
+        env.execute_contract(
+            &QualifiedContractIdentifier::local("contract-b").unwrap(),
+            "as-contract-cc-get-sponsor",
+            &vec![],
+            false
+        )
+        .unwrap(),
+        Value::list_from(vec![sponsor.clone()]).unwrap()
+    );
+}
+
+fn test_tx_sponsor(owned_env: &mut OwnedEnvironment) {
+    let contract_a = "(define-read-only (get-sponsor)
```

Comment:
You should also have a test that covers code executed during the contract initialization. For example:

```
(define-read-only (get-sponsor)
   (list tx-sponsor?))
(asserts! (is-eq tx-sponsor? (some 'SM2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQVX8X0G)) (err 1))
```

What this will do is that when the contract is initialized, it'll see if the contract publish transaction's tx sponsor was equal to `SM2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQVX8X0G`, and if not, will cause a runtime error during the contract initialization.